// @livekit/components-react/dist/hooks.mjs
import { D as e, S as t, w as i, b as u, u as n, i as o, a as r, E as c, c as p, F as P, d, n as l, m as T, G as k, Y as m, H as g, C as S, A as I, e as C, y as A, s as M, o as b, X as f, W as y, I as R, J as V, l as h, T as v, K as D, x as L, B as x, L as B, M as F, v as U, N as w, O as E, f as G, g as K, p as N, Z as Q, P as W, Q as H, k as J, h as O, U as X, R as Y, t as Z, r as j, V as q } from "./hooks-BbVRVxyG.mjs";
import { u as _ } from "./room-FqAH6aXw.mjs";
export {
  e as useAudioPlayback,
  t as useAudioWaveform,
  i as useChat,
  u as useChatToggle,
  n as useClearPinButton,
  o as useConnectionQualityIndicator,
  r as useConnectionState,
  c as useDataChannel,
  p as useDisconnectButton,
  P as useFacingMode,
  d as useFocusToggle,
  l as useGridLayout,
  T as useIsEncrypted,
  k as useIsMuted,
  m as useIsRecording,
  g as useIsSpeaking,
  _ as useLiveKitRoom,
  S as useLocalParticipant,
  I as useLocalParticipantPermissions,
  C as useMediaDeviceSelect,
  A as useMediaDevices,
  M as useMultibandTrackVolume,
  b as usePagination,
  f as useParticipantAttribute,
  y as useParticipantAttributes,
  R as useParticipantInfo,
  V as useParticipantPermissions,
  h as useParticipantTile,
  v as useParticipantTracks,
  D as useParticipants,
  L as usePersistentUserChoices,
  x as usePinnedTracks,
  B as useRemoteParticipant,
  F as useRemoteParticipants,
  U as useRoomInfo,
  w as useSortedParticipants,
  E as useSpeakingParticipants,
  G as useStartAudio,
  K as useStartVideo,
  N as useSwipe,
  Q as useTextStream,
  W as useToken,
  H as useTrackByName,
  J as useTrackMutedIndicator,
  O as useTrackToggle,
  X as useTrackTranscription,
  Y as useTrackVolume,
  Z as useTracks,
  j as useVisualStableUpdate,
  q as useVoiceAssistant
};
