// @livekit/components-react/dist/index.mjs
import { D as s, S as t, w as o, b as n, u as i, i as u, a as r, E as c, c as C, F as d, d as l, n as p, m as P, G as T, Y as k, H as I, C as m, A as x, e as L, y, s as g, o as f, X as S, W as R, I as M, J as h, l as v, T as A, K as b, x as B, B as D, L as V, M as E, v as F, N as w, O as Q, f as N, g as G, p as U, Z as K, P as z, Q as J, k as W, h as j, U as q, R as H, t as O, r as X, V as Y } from "./hooks-BbVRVxyG.mjs";
import { L as _, u as $ } from "./room-FqAH6aXw.mjs";
import { AudioConference as ea, Chat as sa, ControlBar as ta, MediaDeviceMenu as oa, PreJoin as na, VideoConference as ia, VoiceAssistantControlBar as ua, usePreviewDevice as ra, usePreviewTracks as ca } from "./prefabs.mjs";
import { L as da, ab as la, af as pa, R as Pa, ag as Ta, a1 as ka, au as Ia, at as ma, aq as xa, ak as La, F as ya, c as ga, u as fa, i as Sa, ah as Ra, a as Ma, j as ha, B as va, w as Aa, aa as ba, av as Ba, f as Da, aw as Va } from "./contexts-VkyY-dAg.mjs";
import { s as Fa, A as wa, B as Qa, z as Na, E as Ga, g as Ua, S as Ka, a as za, c as Ja, C as Wa, H as ja, n as qa, r as Ha, o as Oa, i as Xa, D as Ya, h as Za, F as _a, p as $a, I as ae, d as ee, G as se, L as te, e as oe, J as ne, M as ie, K as ue, N as re, k as ce, x as Ce, v as de, t as le, b as pe, P as Pe, O as Te, Q as ke, U as Ie, W as me, R as xe, w as Le, X as ye, Y as ge, Z as fe, q as Se, f as Re, l as Me, j as he, u as ve, y as Ae, T as be, _ as Be, V as De, m as Ve } from "./components-DJquvzzg.mjs";
export {
  ea as AudioConference,
  Fa as AudioTrack,
  wa as AudioVisualizer,
  Qa as BarVisualizer,
  Na as CameraDisabledIcon,
  Ga as CameraIcon,
  Ua as CarouselLayout,
  sa as Chat,
  Ka as ChatCloseIcon,
  za as ChatEntry,
  Ja as ChatIcon,
  Wa as ChatToggle,
  ja as Chevron,
  qa as ClearPinButton,
  Ha as ConnectionQualityIndicator,
  Oa as ConnectionState,
  Xa as ConnectionStateToast,
  ta as ControlBar,
  Ya as DisconnectButton,
  Za as FocusLayout,
  _a as FocusLayoutContainer,
  $a as FocusToggle,
  ae as FocusToggleIcon,
  ee as GearIcon,
  se as GridLayout,
  da as LKFeatureContext,
  la as LayoutContext,
  te as LayoutContextProvider,
  oe as LeaveIcon,
  _ as LiveKitRoom,
  ne as LockLockedIcon,
  oa as MediaDeviceMenu,
  ie as MediaDeviceSelect,
  ue as MicDisabledIcon,
  re as MicIcon,
  ce as ParticipantAudioTile,
  pa as ParticipantContext,
  Ce as ParticipantContextIfNeeded,
  de as ParticipantLoop,
  le as ParticipantName,
  pe as ParticipantPlaceholder,
  Pe as ParticipantTile,
  na as PreJoin,
  Te as QualityExcellentIcon,
  ke as QualityGoodIcon,
  Ie as QualityPoorIcon,
  me as QualityUnknownIcon,
  xe as RoomAudioRenderer,
  Pa as RoomContext,
  Le as RoomName,
  ye as ScreenShareIcon,
  ge as ScreenShareStopIcon,
  fe as SpinnerIcon,
  Se as StartAudio,
  Re as StartMediaButton,
  Me as Toast,
  he as TrackLoop,
  ve as TrackMutedIndicator,
  Ta as TrackRefContext,
  Ae as TrackRefContextIfNeeded,
  be as TrackToggle,
  Be as UnfocusToggleIcon,
  ia as VideoConference,
  De as VideoTrack,
  ua as VoiceAssistantControlBar,
  Ve as formatChatMessageLinks,
  ka as isTrackReference,
  Ia as setLogExtension,
  ma as setLogLevel,
  s as useAudioPlayback,
  t as useAudioWaveform,
  o as useChat,
  n as useChatToggle,
  i as useClearPinButton,
  u as useConnectionQualityIndicator,
  r as useConnectionState,
  xa as useCreateLayoutContext,
  c as useDataChannel,
  C as useDisconnectButton,
  La as useEnsureCreateLayoutContext,
  ya as useEnsureLayoutContext,
  ga as useEnsureParticipant,
  fa as useEnsureRoom,
  Sa as useEnsureTrackRef,
  d as useFacingMode,
  Ra as useFeatureContext,
  l as useFocusToggle,
  p as useGridLayout,
  P as useIsEncrypted,
  T as useIsMuted,
  k as useIsRecording,
  I as useIsSpeaking,
  Ma as useLayoutContext,
  $ as useLiveKitRoom,
  m as useLocalParticipant,
  x as useLocalParticipantPermissions,
  ha as useMaybeLayoutContext,
  va as useMaybeParticipantContext,
  Aa as useMaybeRoomContext,
  ba as useMaybeTrackRefContext,
  L as useMediaDeviceSelect,
  y as useMediaDevices,
  g as useMultibandTrackVolume,
  f as usePagination,
  S as useParticipantAttribute,
  R as useParticipantAttributes,
  Ba as useParticipantContext,
  M as useParticipantInfo,
  h as useParticipantPermissions,
  v as useParticipantTile,
  A as useParticipantTracks,
  b as useParticipants,
  B as usePersistentUserChoices,
  D as usePinnedTracks,
  ra as usePreviewDevice,
  ca as usePreviewTracks,
  V as useRemoteParticipant,
  E as useRemoteParticipants,
  Da as useRoomContext,
  F as useRoomInfo,
  w as useSortedParticipants,
  Q as useSpeakingParticipants,
  N as useStartAudio,
  G as useStartVideo,
  U as useSwipe,
  K as useTextStream,
  z as useToken,
  J as useTrackByName,
  W as useTrackMutedIndicator,
  Va as useTrackRefContext,
  j as useTrackToggle,
  q as useTrackTranscription,
  H as useTrackVolume,
  O as useTracks,
  X as useVisualStableUpdate,
  Y as useVoiceAssistant
};
